include/tlapack/lapack/pbtrf_with_workspace.hpp

```diff
@@ -0,0 +1,223 @@
+/// @file trmm_out.hpp
+/// @author Ella Addison-Taylor, Kyle Cunningham, University of Colorado Denver,
+/// USA
+//
+// Copyright (c) 2025, University of Colorado Denver. All rights reserved.
+//
+// This file is part of <T>LAPACK.
+// <T>LAPACK is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#ifndef TLAPACK_PBTRF_WITH_WORKSPACE_HH
+#define TLAPACK_PBTRF_WITH_WORKSPACE_HH
+
+#include <tlapack/blas/trsm.hpp>
+#include <tlapack/lapack/potf2.hpp>
+#include <tlapack/lapack/mult_llh.hpp>
+#include <tlapack/lapack/mult_uhu.hpp>
+
+namespace tlapack {
+
+/// @brief Options struct for pbtrf_with_workspace()
+struct BlockedBandedCholeskyOpts : public EcOpts {
+    constexpr BlockedBandedCholeskyOpts(const EcOpts& opts = {})
+        : EcOpts(opts) {};
+
+    size_t nb = 32;  // Block size
+};
+/** 
+ * 
+ * Cholesky factorization of a full, banded, n by n matrix. 
+ *
+ * @param[in] uplo
+ *     What part of the matrix A is referenced,
+ *     the opposite band being assumed to be zero:
+ *     - Uplo::Lower: A is lower banded.
+ *     - Uplo::Upper: A is upper banded.
+ *     - Uplo::General: A would be treated as lower banded.
+ *
+ * @param[in, out] A
+ *     A banded, full, n by n Hermitian matrix
+ *
+ * @param[in] kd The band size.
+ *
+ * @param[in] opts Options.
+ *    - Define the size of the block, nb.
+ *    - Default is nb = 32.
+ *
+ * @ingroup auxiliary
+ **/
+
+template<typename uplo_t, typename matrix_t>
+void pbtrf_with_workspace(uplo_t uplo, matrix_t& A, size_t kd, const BlockedBandedCholeskyOpts& opts = {})
+ {
+     using T = tlapack::type_t<matrix_t>;
+     using idx_t = tlapack::size_type<matrix_t>;
+     using range = tlapack::pair<idx_t, idx_t>;
+     using real_t = tlapack::real_type<T>;
+
+     tlapack::Create<matrix_t> new_matrix;
+
+     const idx_t nb = opts.nb;
+
+     idx_t n = nrows(A);
+
+     std::vector<T> work_(nb * nb);
```

Comment:
(COMMENT, NO ACTION REQUIRED, READ and CLICK "resolve")
Good with me to have memory allocation within the code for now.
Mostly because we have an algorithm to remove memory allocation. So great, this will go away.
If we keep a workspace code, then we will have to have a memory allocation layer. (See other \<T\>LAPACK codes.)